packages/chain-adapters/src/utils/utxoUtils.ts

```diff
@@ -122,16 +139,23 @@ const accountTypeToVersion = {
   [UtxoAccountType.SegwitNative]: Buffer.from(PublicKeyType.zpub, 'hex')
 }
 
+const convertVersions = ['xpub', 'ypub', 'zpub']
+
 /**
  * Convert any public key into an xpub, ypub, or zpub based on account type
  *
  * Blockbook generates addresses from a public key based on the version bytes
  * some wallets always return the public key in "xpub" format, so we need to convert those
  *
+ * TODO - why is this here/used? we shouldn't need to convert xpubs afaik
```

Comment:
see the comment above in regards to how blockbook handles pubkeys, ensures that the pubkey returned matches the prefix format as determined by blockbook for proper account lookup (i believe, or is my best understanding)

Reply:
blockbook follows standards - xpub/ypub/zpub for p2pkh, p2sh-p2wpkh, and p2wpkh. ltub for litecoin, gdub for doge. the version bytes are a standard thing, we should never be converting between them unless we're obtaining an extended pubkey from a 3rd party/source beyond our control. legacy (beta) did this all wrong to accommodate an invalid implementation in WT.

Reply:
yeah for sure, I think that is the "some wallets" case. Not sure this was added by @mrnerdhair I believe who could probably explain his thought process better

Reply:
https://github.com/shapeshift/lib/commit/624e13a1fc3a4e770ef212100919baa922b365bc
added by @cjthompson